pkg/shp/flags/buildrun.go

```diff
@@ -0,0 +1,49 @@
+package flags
+
+import (
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/spf13/pflag"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BuildRunSpecFlags BuildRun's spec represtantation as command-line flags.
+func BuildRunSpecFlags(flags *pflag.FlagSet) *buildv1alpha1.BuildRunSpec {
+	spec := &buildv1alpha1.BuildRunSpec{
+		BuildRef:       &buildv1alpha1.BuildRef{},
+		ServiceAccount: &buildv1alpha1.ServiceAccount{},
+		Timeout:        &metav1.Duration{},
+		Output: &buildv1alpha1.Image{
+			Credentials: &corev1.LocalObjectReference{},
+		},
+	}
+
+	buildRefFlags(flags, spec.BuildRef)
+	serviceAccountFlags(flags, spec.ServiceAccount)
+	timeoutFlags(flags, spec.Timeout)
+	imageFlags(flags, "output", spec.Output)
+
+	return spec
+}
+
+// SanitizeBuildRunSpec when inner elements are empty, making sure they are replace by a nil pointer.
```

Comment:
nit - clean up grammar in godoc and make it consistent with SanitizeBuildSpec.

```suggestion
// SanitizeBuildRunSpec checks for empty inner data structures and replaces them with nil.
```